### DISCUSSION TITLE
Can I use nuget packages for Xamarin.Forms which do not have Xamarin.Forms dependency like Plugin.AudioRecorder?

### DISCUSSION THREAD

**Discussion by ivoryguard** [AvaloniaUI/Avalonia#15550] [Q&A]
hello.

There are nuget packages for Xamarin.Forms, which are UI-less packages with no dependencies on Xamarin.Forms. 

For example, Plugin.AudioRecorder corresponds to Xamarin.Forms, but it does not have any dependencies on Xamarin.Forms.

Can I use these UI-less packages for Xamarin.Forms in Avalonia crossplatform for mobile project?

If so, I would like to know the specific dependency restrictions.

Since Avalonia performs UI rendering only, I'm not sure how to struct cross-platform project and package reference when trying to use hardware/native resources like audio or video.

Best regards.

**Comment by timunie** [comment]
Xamarin.Essentails or Maui.Essentials can be used, yes. If it is Xamarin.Forms I'm not sure, but worth a test.

**Comment by maxkatz6** [comment]
[Plugin.AudioRecorder](https://github.com/NateRickard/Plugin.AudioRecorder) is not maintained for five years now. But you might find a fork that support modern .NET 8 targets.

**Reply by maxkatz6** [reply]
Or you can call these platforms APIs directly from your platform specific projects:
https://github.com/NateRickard/Plugin.AudioRecorder/blob/master/Plugin.AudioRecorder.iOS/
https://github.com/NateRickard/Plugin.AudioRecorder/tree/master/Plugin.AudioRecorder.Android/
